src/js/views/programs/sidebar/action/action-sidebar_views.js

```diff
@@ -286,20 +296,51 @@ const LayoutView = View.extend({
     this.showChildView('details', new DetailsView({ model: this.model, action: this.action }));
   },
   showPublished() {
+    const published = !!this.action.get('published_at');
+    const isDisabled = this.action.isNew();
+
+    const toggleView = new ToggleComponent({
+      status: published,
+      isDisabled,
+    });
+
+    this.listenTo(toggleView, 'click', () => {
+      const newPublishedAt = published ? null : dayjs.utc().format();
+      this.action.save({ published_at: newPublishedAt });
+    });
+
+    this.showChildView('published', toggleView);
+  },
+  showArchived() {
+    const archived = !!this.action.get('archived_at');
+    const isDisabled = this.action.isNew();
+
+    const toggleView = new ToggleComponent({
+      status: archived,
+      isDisabled,
+    });
+
+    this.listenTo(toggleView, 'click', () => {
+      const newArchivedAt = archived ? null : dayjs.utc().format();
+      this.action.save({ archived_at: newArchivedAt });
```

Comment:
This currently doesn't work. `program-action.archived_at` isn't being set in the db when I test this locally. It's always `null`, regardless of whether I send `archived_at: null` or `archived_at: utc_date` in the `PATCH` api request.

It's the same method I'm using for the program and program-flow (which work), so I'm not sure what's up. Either a bug by me or something on the backend side of things.

Reply:
I'm *pretty sure* I'm doing everything correct for this.

I'm sending this data with the `/api/program-actions/:id` `PATCH` request:

```js
{
    "data": {
        "id": "46f30482-62a8-11ec-8efa-b558ead68b59",
        "type": "program-actions",
        "attributes": {
            "archived_at": "2023-10-13T16:39:44Z"
        }
    }
}
```

But always get `archived_at: null` back from the backend for that action.

I think it's a backend issue.

Reply:
@nmajor25 do you have an up to date `develop` checked out locally? The code is definitely in place:

<img width="655" alt="image" src="https://github.com/RoundingWell/care-ops-frontend/assets/38203/f5891120-4ded-40ce-af48-c8203941955d">

Reply:
Yup, I do. And I also did a `./bin/docker-rebuild` just to be sure.

Reply:
@nmajor25 hmmm, okay. I know you're off this afternoon, but make a bug card for me when you get a chance, please.

Reply:
@shadowhand  I just created that story here: [sc-42170](https://app.shortcut.com/roundingwell/story/42170/unable-to-set-archived-at-on-program-actions).